scipy/linalg/tests/test_decomp.py

```diff
@@ -653,6 +653,25 @@ def test_eigh():
                                    turbo, eigenvalues)
 
 
+def test_eigh_rng():
+    DIM = 6
+    v = {'dim': (DIM,),
+         'dtype': ('f','d','F','D'),
+         'overwrite': (True, False),
+         'lower': (True, False),
+         'eigrng': (None, (-.5, .5), (-.75, .75))}
+
+    for dim in v['dim']:
+        for typ in v['dtype']:
+            for overwrite in v['overwrite']:
+                for eigenrange in v['eigrng']:
+                    for lower in v['lower']:
```

Comment:
Note that itertools.product can be used to avoid deeply nested loops:
```
dims = (DIM,)
dtypes = ('f', 'd', 'F', 'D')
...
parts = [dims, dtypes, ...]
for dim, typ, overwrite, eigenrange, lower in itertools.product(*parts):
    yield (...)
```